src/main/java/org/testng/Assert.java

```diff
@@ -152,13 +152,13 @@ private static boolean areNotEqualImpl(Object actual, Object expected) {
     if ((expected == null) && (actual == null)) {
       return false;
     }
-    if (expected == null ^ actual == null) {
-      return true;
+    if (expected == null) {
+      return actual.equals(actual);
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Self comparison of value with itself (SA_LOCAL_SELF_COMPARISON)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-comparison-of-value-with-itself-sa-local-self-comparison).
This method compares a local variable with itself, and may indicate a typo or a logic error. Make sure that you are comparing the right things.

Reply:
Thanks @ellieMayVelasquez, but these rules don't have any sense in this very specific context ;)

Reply:
noticed